src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -0,0 +1,86 @@
+package com.github.sdonkov;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class PrefixTreeTest {
+
+    private static List<CharSequence> words;
+    private static PrefixTree prefixTree;
+
+    @BeforeAll
+    static void setUp() throws IOException {
+        words = new ArrayList<>();
+        prefixTree = new PrefixTree();
+        try (InputStream is = PrefixTreeTest.class.getResourceAsStream("/tests.txt")) {
+            if (is != null) {
```

Comment:
Here and below, there is usually no reason to write test code defensively. On the contrary, this may lead to passing tests when in reality things have messed up. If `is` is `null`, it means we can't test, i.e. we need the crash.